content/200-orm/200-prisma-client/500-deployment/301-edge/450-deploy-to-cloudflare.mdx

```diff
@@ -536,6 +545,193 @@ npx wrangler deploy
+-- CreateTable
+CREATE TABLE "User" (
+    "id" INTEGER NOT NULL PRIMARY KEY AUTOINCREMENT,
+    "email" TEXT NOT NULL,
+    "name" TEXT
+);
+
+-- CreateIndex
+CREATE UNIQUE INDEX "User_email_key" ON "User"("email");
+```
+
+You now need to use the [`wrangler d1 migrations apply`](https://developers.cloudflare.com/workers/wrangler/commands/#migrations-apply) command to send this SQL statement to D1. Note that this command accepts two options:
+
+- `--local`: Executes the statement against a _local_ version of D1. This local version of D1 is a SQLite database file that'll be located in your project. This approach is useful, when you want to develop and test your Worker on your local machine. Learn more in the [Cloudflare docs](https://developers.cloudflare.com/d1/configuration/local-development/).
+- `--remote`: Executes the statement against your _remote_ version of D1. This version is used by your _deployed_ Cloudflare Workers. Learn more in the [Cloudflare docs](https://developers.cloudflare.com/d1/configuration/remote-development/).
+
+In this tutorial, you'll do both: test the Worker locally _and_ deploy it afterwards. So, you need to run both commands. Open your terminal and paste the following commands:
+
+```terminal
+# For the local database
+npx wrangler d1 migrations apply --local
+
+# For the remote database
+npx wrangler d1 migrations apply --remote
+```
+
+Let's also create some dummy data that we can query once the Worker is running. This time, you'll run the SQL statement without storing it in a file:
+
+```terminal
+# For the local database
+npx wrangler d1 execute prisma-demo-db --command "INSERT INTO  \"User\" (\"email\", \"name\") VALUES
+('jane@prisma.io', 'Jane Doe (Local)');" --local
+
+# For the remote database
+npx wrangler d1 execute prisma-demo-db --command "INSERT INTO  \"User\" (\"email\", \"name\") VALUES
+('jane@prisma.io', 'Jane Doe (Remote)');" --remote
+```
+
+#### 5. Use Prisma Client in your Worker to send a query to the database
+
+In order to query your database from the Worker using Prisma ORM, you need to:
+
+1. Add the `DB` to the `Env` interface.
+2. Instantiate `PrismaClient` using the `PrismaD1` driver adapter.
+3. Send a query using Prisma Client and return the result.
+
+Open `src/index.ts` and replace the entire content with the following:
+
+```typescript file=src/index.ts
+import { PrismaClient } from '@prisma/client'
+import { PrismaD1 } from '@prisma/adapter-d1'
+
+export interface Env {
+  DB: D1Database
+}
+
+export default {
+  async fetch(
+    request: Request,
+    env: Env,
+    ctx: ExecutionContext
+  ): Promise<Response> {
+    const adapter = new PrismaD1(env.DB)
+    const prisma = new PrismaClient({ adapter })
+
+    const users = await prisma.user.findMany()
+    const result = JSON.stringify(users)
+    return new Response(result)
+  },
+}
+```
+
+Before running the Worker, you need to generate Prisma Client with the following command:
+
+```
+npx prisma generate
+```
```

Comment:
I think we should first tell the users to run `npx prisma generate` (which makes TypeScript's autocompletions happy), and then display the TypeScript snippet.
So, the numerical steps above could become:

1. Add the `DB` to the `Env` interface.
2. Generate `@prisma/client` types.
3. Instantiate `PrismaClient` using the `PrismaD1` driver adapter.
4. Send a query using Prisma Client and return the result.

Reply:
Good call!! There was an instruction to generate Prisma Client _after_ the code snippet but I now moved it before: 
<img width="475" alt="image" src="https://github.com/prisma/docs/assets/4058327/d48a17d0-3ad5-4b23-8777-a8ca2bfbd411">

I don't want to include that instruction in the list because that list is all about _code changes_ while this one is an instruction to run a terminal command to generate PC.